packages/next/src/server/future/route-modules/app-route/helpers/auto-implement-methods.ts

```diff
@@ -34,8 +34,8 @@ export function autoImplementMethods(
     // exists).
     if (method === 'HEAD') {
       // If the userland module doesn't implement the GET method, then
-      // we're done.
-      if (!handlers.GET) break
+      // we can continue to check the OPTION method below.
+      if (!handlers.GET) continue
```

Comment:
Let's move this to the `if` condition to make it more clear and clarify this is just re-using GET for HEAD if available.